src/workspace/mod.rs

```diff
@@ -0,0 +1,2406 @@
+          config_url: member_config.specifier.clone(),
+          kind: WorkspaceDiagnosticKind::RootOnlyOption("nodeModulesDir"),
+        });
+      }
+      if member_config.json.scopes.is_some() {
+        diagnostics.push(WorkspaceDiagnostic {
+          config_url: member_config.specifier.clone(),
+          kind: WorkspaceDiagnosticKind::RootOnlyOption("scopes"),
+        });
+      }
+      if !member_config.json.unstable.is_empty() {
+        diagnostics.push(WorkspaceDiagnostic {
+          config_url: member_config.specifier.clone(),
+          kind: WorkspaceDiagnosticKind::RootOnlyOption("unstable"),
+        });
+      }
+      if member_config.json.vendor.is_some() {
+        diagnostics.push(WorkspaceDiagnostic {
+          config_url: member_config.specifier.clone(),
+          kind: WorkspaceDiagnosticKind::RootOnlyOption("vendor"),
+        });
+      }
+      if member_config.json.workspace.is_some() {
+        diagnostics.push(WorkspaceDiagnostic {
+          config_url: member_config.specifier.clone(),
+          kind: WorkspaceDiagnosticKind::RootOnlyOption("workspace"),
+        });
+      }
+      if let Some(value) = &member_config.json.lint {
+        if value.get("report").is_some() {
+          diagnostics.push(WorkspaceDiagnostic {
+            config_url: member_config.specifier.clone(),
+            kind: WorkspaceDiagnosticKind::RootOnlyOption("lint.report"),
+          });
+        }
+      }
+      if let Some(name) = member_config.json.name.as_deref() {
+        if let Some(other_member_url) = seen_names.get(name) {
+          diagnostics.push(WorkspaceDiagnostic {
+            config_url: member_config.specifier.clone(),
+            kind: WorkspaceDiagnosticKind::DuplicateMemberName {
+              name: name.to_string(),
+              other_member: (*other_member_url).clone(),
+            },
+          });
+        } else {
+          seen_names.insert(name, &member_config.specifier);
+        }
+      }
+    }
+
+    fn check_all_configs(
+      config: &ConfigFile,
+      diagnostics: &mut Vec<WorkspaceDiagnostic>,
+    ) {
+      if config.json.deprecated_workspaces.is_some() {
+        diagnostics.push(WorkspaceDiagnostic {
+          config_url: config.specifier.clone(),
+          kind: WorkspaceDiagnosticKind::InvalidWorkspacesOption,
+        });
+      }
+    }
+
+    let mut diagnostics = Vec::new();
+    let is_deno_workspace = self
+      .root_folder()
+      .1
+      .deno_json
+      .as_ref()
+      .map(|d| d.json.workspace.is_some())
+      .unwrap_or(false);
+    let mut seen_names = HashMap::with_capacity(self.config_folders.len());
+    for (url, folder) in &self.config_folders {
+      if let Some(config) = &folder.deno_json {
+        if is_deno_workspace {
+          if url == &self.root_dir {
+            check_root_diagnostics(config, &mut diagnostics);
+          } else {
+            check_member_diagnostics(config, &mut diagnostics, &mut seen_names);
+          }
```

Comment:
Does this account for the `workspace = [".", ...]` case, or is it omitted for now intentionally?

Reply:
I just added code to specifically disallow it referencing itself or doing stuff like `"."` or it referencing itself like `"../self"`. I don't think it makes sense for someone to put that in a config file.

Reply:
Scratch that. I'm changing it to account for `"."` because fresh apparently uses this.

Reply:
Handled in https://github.com/denoland/deno_config/pull/58/commits/6455372c5fa4e4d03c5435cbd62f97026e168acb